### DISCUSSION TITLE
Updated to most recent version and now images won't load

### DISCUSSION THREAD

**Discussion by CallumToddYTL** [dompdf/dompdf#3136] [Q&A]
I didn't change anything but the file directory for DOMPDF in my code. However, since updating to the newest version, I haven't been able to load images. I've tried absolutely everything and nothing seems to be working.

I check my HTML file via URL and the images are loading on there. Is there a new bug with the latest update or is it something I haven't considered. PDF's do generate and there are text values in it. Just every page has a grey cross as seen in the screenshot.

![image](https://user-images.githubusercontent.com/106147214/219338370-3c18cc48-bd3f-488d-96b1-ffb449275519.png)

Other screenshots include the pdf generation and also my file paths for images.

![image](https://user-images.githubusercontent.com/106147214/219338246-08b7a0c5-51f7-4596-ae91-149677dc4031.png)
![image](https://user-images.githubusercontent.com/106147214/219338476-5b292e58-ddab-4f69-93ad-50035e57c9e7.png)
![image](https://user-images.githubusercontent.com/106147214/219338565-38f4274a-86c5-44ac-ad66-8671bc3a11bc.png)

**Comment by bsweeney** [comment]
You didn't mention which version you were using previously so I can't say exactly why the upgrade impacted you in this way. However, right away I see that you have not specified a chroot path. Filesystem resources must be contained within the path(s) specified for this option.

```
$options = new Options();
$options->set("chroot", ["/basic-calc-1.4.0/public_html"]);
```

I'm assuming the "basic-calc-1.4.0" path is at the root of the filesystem. Otherwise that could be part of the issue as well since you have set it as your base path.

**Reply by CallumToddYTL** [reply]
Thanks man.
I specified a chroot path but left it empty and that worked for some reason. Not exactly sure why, but hey-ho. At least it's working now. Thanks for the pointer!

**Reply by bsweeney** [reply]
The default path is the Dompdf directory. Leaving that option empty would remove restrictions on what files Dompdf will load. Most potential information disclosure issues have been squashed, though if users are allowed to specify HTML content local file reads could be used for fingerprinting or to retrieve, for example, images containing sensitive information. That's why we generally recommend a more tightly controlled path.
